packages/cyberstorm/src/components/Toast/Provider.tsx

```diff
@@ -0,0 +1,71 @@
+import { createContext, PropsWithChildren, useReducer } from "react";
+import { v4 as uuid } from "uuid";
+import * as RadixToast from "@radix-ui/react-toast";
+import Toast from ".";
+import { ToastProps } from "./Toast";
+
+const initState: {
+  toasts: ToastProps[];
+} = { toasts: [] };
+
+const toastReducer = (
+  state: {
+    toasts: ToastProps[];
+  },
+  action: {
+    type: "add" | "delete";
+    toast: ToastProps;
+  }
+) => {
+  switch (action.type) {
+    case "add": {
+      return {
+        toasts: [...state.toasts, action.toast],
+      };
+    }
+    case "delete": {
+      const updatedToasts = state.toasts.filter(
+        (toast) => toast.id !== action.toast.id
+      );
+      return {
+        toasts: updatedToasts,
+      };
+    }
+    default:
+      throw new Error(`Unhandled action: ${action.type}`);
+  }
+};
+
+interface ContextInterface {
+  addToast: (props: Omit<ToastProps, "id">) => void;
```

Comment:
Strictly speaking I wouldn't consider these props since they're not passed to JSX component 🤓